test/setup_aot/TestSetupAOT.java

```diff
@@ -44,16 +52,161 @@
-            t.run(System.out, System.out, "--help");
+        runJDKTools(args);
+        streamOps(args);
+        LOGGER.log(Level.FINE, "Done");
+    }
+
+    static void runJDKTools(String[] args) throws Throwable {
+        String tmpDir = args[0];
+        System.out.println("Working Directory = " + System.getProperty("user.dir"));
+        System.out.println("Temp output dir = " + tmpDir);
+
+        // ------------------------------
+        // javac
+
+        execTool("javac", "--help");
+
+        JavacBenchApp.main(new String[] {"5"});
+
+        // ------------------------------
+        // javap
+
+        execTool("javap", "--help");
+        execTool("javap", "-c", "-private", "-v", "-verify",
+                 "java.lang.System",
+                 "java/util/stream/IntStream",
+                 "jdk.internal.module.ModuleBootstrap");
+
+        // ------------------------------
+        // jlink
+
+        String jlinkOutput = tmpDir + File.separator + "jlinkOutput";
+
+        execTool("jlink", "--help");
+        execTool("jlink", "--list-plugins");
+
+        deleteAll(jlinkOutput);
+        execTool("jlink", "--add-modules", "java.base", "--strip-debug", "--output", jlinkOutput);
+        deleteAll(jlinkOutput);
+
+        // ------------------------------
+        // jar
+
+        String jarOutput = tmpDir + File.separator + "tmp.jar";
+
+        execTool("jar", "--help");
+
+        deleteAll(jarOutput);
+        execTool("jar", "cvf", jarOutput, "TestSetupAOT.class");
+        execTool("jar", "uvf", jarOutput, "TestSetupAOT.class");
+        execTool("jar", "tvf", jarOutput);
+        execTool("jar", "--describe-module", "--file=" + jarOutput);
+        deleteAll(jarOutput);
+
+        // ------------------------------
+        // jdeps
+
+        execTool("jdeps", "--help");
+        execTool("jdeps", "-v", "TestSetupAOT.class");
+    }
+
+    static void deleteAll(String f) {
+        deleteAll(new File(f));
+    }
+
+    static void deleteAll(File f) {
+        File[] files = f.listFiles();
+        if (files != null) {
+            for (File file : files) {
+                deleteAll(file);
+            }
         }
+        System.out.println("Deleting: " + f);
+        f.delete();
+    }
+
+    static void execTool(String tool, String... args) throws Throwable {
+        System.out.println("== Running tool ======================================================");
+        System.out.print(tool);
+        for (String s : args) {
+            System.out.print(" " + s);
+        }
+        System.out.println();
+        System.out.println("======================================================================");
+
+        ToolProvider t = ToolProvider.findFirst(tool)
+            .orElseThrow(() -> new RuntimeException(tool + " not found"));
+        t.run(System.out, System.out, args);
+    }
+
+
+    static void streamOps(String args[]) {
```

Comment:
Can you please add a comment with explanation of purpose of 'streamOps' method.

Reply:
I renamed the method to `invokedynamicTests()` and added comments.